data/account/participationRegistry.go

```diff
@@ -332,61 +415,176 @@ func (db *participationDB) GetAll() (records []ParticipationRecord, err error) {
 	return
 }
 
+func (db *participationDB) Get(id ParticipationID) (record ParticipationRecord, err error) {
+	db.mutex.RLock()
+	defer db.mutex.RUnlock()
+
+	record, ok := db.cache[id]
+	if !ok {
+		return ParticipationRecord{}, ErrParticipationIDNotFound
+	}
+	return record, nil
+}
+
+func (db *participationDB) GetAll() ([]ParticipationRecord, error) {
+	db.mutex.RLock()
+	defer db.mutex.RUnlock()
+
+	results := make([]ParticipationRecord, 0, len(db.cache))
+	for _, record := range db.cache {
+		copy := record
```

Comment:
This would not perform a deep copy, if this was your intent.

Reply:
It is my intent to create a copy, I don't want the caller to modify the cache.

Are you thinking that if there were pointer values in the cache this would fail? In this case, `ParticipationRecord` has no pointers, but is there a safer way to ensure a deep copy?